src/Microsoft.Framework.ApplicationHost/Impl/Syntax/CommandGrammar.cs

```diff
@@ -29,7 +29,9 @@ public CommandGrammar(Func<string, string> variable)
 
             var unquotedTerm = Rep1(unquotedPiece.Or(specialPiece)).Str();
 
-            var quotedTerm = Ch('\"').And(Rep(quotedPiece.Or(specialPiece)).Str()).And(Ch('\"')).Left().Down();
+            var quotedTerm = Ch('\"').And(Rep(quotedPiece.Or(specialPiece)).Str()).And(Ch('\"'))
+                .Left().Down()
+                .Build(str => "\"" + str + "\"");
```

Comment:
As @davidfowl and I discussed offline, will make restoration of surrounding quotes conditional. Won't be done when parsing `commands` but will be for `scripts`.